### PR TITLE
Structured code summarization

### PR DESCRIPTION
**Pull request by alekszievr**
## Summary by CodeRabbit

- **New Features**
	- Introduced a new function for running the code graph pipeline.
	- Added a new file with instructions for summarizing Python code.
	- Added a new function to summarize code content asynchronously.
	- Introduced new data model classes for summarized code components.

- **Bug Fixes**
	- Improved error handling in the pipeline execution process.

- **Documentation**
	- New instructions for summarizing Python code to aid technical writers and programmers.

- **Refactor**
	- Updated function signatures and removed deprecated parameters for better usability.

### CODE REVIEW

File: cognee/api/v1/cognify/code_graph_pipeline.py [hunk 1]
```diff
@@ -129,7 +136,7 @@ async def run_code_graph_pipeline(repo_path):
         Task(get_repo_file_dependencies),
         Task(enrich_dependency_graph, task_config={"batch_size": 50}),
         Task(expand_dependency_graph, task_config={"batch_size": 50}),
-        Task(summarize_code, summarization_model=SummarizedContent, task_config={"batch_size": 50}),
+        Task(summarize_code, summarization_model=SummarizedCode, task_config={"batch_size": 50}),
```
Comment (lxobr) [inline cognee/api/v1/cognify/code_graph_pipeline.py:139]: Should the the summarize_code function have a summarization_model argument (as it seems to be always called with the SummarizedCode pydantic model)?

File: cognee/modules/data/extraction/extract_summary.py [hunk 2]
```diff
@@ -11,3 +14,12 @@ async def extract_summary(content: str, response_model: Type[BaseModel]):
     llm_output = await llm_client.acreate_structured_output(content, system_prompt, response_model)
 
     return llm_output
+
+async def extract_code_summary(content: str, response_model: Type[BaseModel]):
+    llm_client = get_llm_client()
+
+    system_prompt = read_query_prompt("summarize_code.txt")
```
[the remark this answers is not in the document]
Comment (alekszievr) [reply cognee/modules/data/extraction/extract_summary.py:21]: ah right, file added now!

File: cognee/api/v1/cognify/code_graph_pipeline.py (hunk 1 above)
Comment (alekszievr) [reply cognee/api/v1/cognify/code_graph_pipeline.py:139]: Fair point, I can't think of a reason to keep it, so I'll remove it.

Comment (alekszievr) [reply cognee/api/v1/cognify/code_graph_pipeline.py]: done

**Review by lxobr — APPROVED** [review]

**Pull request merged**